examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,149 @@
+"""
+Title: Image Enhancer/Denoiser using ESPCN
+Author: Anish B
+Date created: 31/12/2021
+Last modified: 05/01/2022
+Description: Implementation of Simple Image Enhancer/Denoiser using Sub-Pixel CNN using U-Net Architecture.
+"""
+import numpy as np
+from glob import glob
+import tensorflow as tf
+import matplotlib.pyplot as plt
+from tensorflow.keras import Model
+from tensorflow.keras import layers
+from tensorflow.keras.applications import ResNet50V2
+from tensorflow.keras.callbacks import EarlyStopping, TensorBoard, ModelCheckpoint
+
+"""shell
+!gdown https://drive.google.com/uc?id=18u1IlpaMuBtMgKMEYHLTZWxVYMA5RQE7
+!unzip -q data.zip
+"""
+
+train_source = sorted(glob("data/source/*.jpg"))[:-60]
+train_targets = sorted(glob("data/target/*.jpg"))[:-60]
+
+val_source = sorted(glob("data/source/*.jpg"))[-60:]
+val_targets = sorted(glob("data/target/*.jpg"))[-60:]
+
+height, width = 352, 608
+
+
+def resize_with_pad(image, h, w):
+    return tf.image.resize_with_pad(image=image, target_height=h, target_width=w)
+
+
+def data_preprocess(source, target):
+    source_image = tf.io.decode_png(tf.io.read_file(source), channels=3)
+    target_image = tf.io.decode_png(tf.io.read_file(target), channels=3)
+    source_image = resize_with_pad(source_image, h=height, w=width) / 255.0
```

Comment:
Since you use a pretrained model you should use the corresponding `preprocess_input` method from `keras.applications`